tests/math_builtin_api/modules/test_generator.py

```diff
@@ -118,22 +141,51 @@ def generate_function_call(sig, arg_names, arg_src):
     fc = fc[:-1] + ");\n"
     return fc
 
+def generate_function_private_call(sig, arg_names, arg_src, types):
+    fc = arg_src
+    fc += "$RETURN_TYPE res = " + sig.namespace + "::" + sig.name + "("
+    for arg_n in arg_names:
+        fc += arg_n + ","
+    fc = fc[:-1] + ");\n"
+    fc += "return privatePtrCheck<" + sig.ret_type + ","
+    fc += extract_type(types[sig.arg_types[-1]]).name + ">("
+    fc += "res, multiPtrSourceData);\n"
+    return fc
+
 def generate_reference(sig, arg_names, arg_src):
     fc = arg_src
-    fc += "$RETURN_TYPE ref = reference::" + sig.name + "("
+    fc += "sycl_cts::resultRef<$RETURN_TYPE> ref = reference::" + sig.name + "("
     for arg_n in arg_names:
         fc += arg_n + ","
     fc = fc[:-1] + ");\n"
     return fc
 
+def generate_reference_ptr(types, sig, arg_names, arg_src):
+    fc = arg_src[:arg_src.rfind('\n', 0, -1)] + "\n"
```

Comment:
I have some issues understanding this... Are you doing a kind of `std.rstrip()`?